code/modules/mob/living/basic/lavaland/basilisk/basilisk_overheat.dm

```diff
@@ -0,0 +1,79 @@
+/// Status effect gained by basilisks when they touch something hot
+/datum/status_effect/basilisk_overheat
+	id = "basilisk_overheat"
+	duration = 3 MINUTES
+	/// Things which will chill us out if we get hit by them
+	var/static/list/chilling_reagents = list(
+		/datum/reagent/medicine/cryoxadone,
+		/datum/reagent/firefighting_foam,
+		/datum/reagent/consumable/frostoil,
+		/datum/reagent/consumable/ice,
+		/datum/reagent/water,
+	)
+
+/datum/status_effect/basilisk_overheat/on_creation(mob/living/new_owner, ...)
+	if (!istype(new_owner, /mob/living/basic/mining/basilisk))
+		return FALSE // Behaviour here is too specific to be reused
+	return ..()
+
+/datum/status_effect/basilisk_overheat/on_apply()
+	. = ..()
```

Comment:
I think you need to check `if(owner.stat != CONSCIOUS) return FALSE`, from what I can tell it can "heat up" if you, say, try to welder a dead Basilisk